File: src/main/java/komu/demodel/domain/InputSource.java
/*
 * Copyright (C) 2010 Juha Komulainen. All rights reserved.
 */
package komu.demodel.domain;

import java.io.IOException;

import komu.demodel.utils.ResourceProvider;

public interface InputSource {
    ResourceProvider getResources() throws IOException;
}

File: src/main/scala/komu/demodel/domain/project/InputSource.scala
/*
 * Copyright (C) 2010 Juha Komulainen. All rights reserved.
 */
package komu.demodel.domain.project

import komu.demodel.utils.Resource

trait InputSource {
  def withResources(thunk: Resource => Unit)
  
  def mapResources[T](thunk: Resource => T): Iterable[T] = {
    val result = new scala.collection.mutable.ArrayBuffer[T]
    withResources(r => result += thunk(r))
    result
  }
}
